admin/services/pupil-upload.service.js

```diff
@@ -0,0 +1,71 @@
+const csv = require('fast-csv')
+const moment = require('moment')
+const fs = require('fs-extra')
+const uuidv4 = require('uuid/v4')
+const { promisify } = require('bluebird')
+const { azureUploadFile } = require('./data-access/azure-file.data.service')
+const csvValidator = require('../lib/validator/csv-validator')
+const singlePupilValidation = require('./single-pupil-validation.service')
+const { insertMany } = require('./data-access/insert-pupils.data.service')
+
+module.exports.upload = async (school, uploadFile) => {
+  let stream
+  const pr = await new Promise((resolve) => {
+    let csvData = []
+    stream = fs.createReadStream(uploadFile.file)
+    csv.fromStream(stream)
+      .on('data', (data) => {
+        csvData.push(data)
+      })
+      .on('end', async () => {
+        // Remove error column and headers from data and validate rows
```

Comment:
This entire block is rather long, so it would be better if it was extracted out into another method.